client/interface.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"context"
+	"encoding"
+	"time"
+)
+
+// Client represents the drand Client interface.
+type Client interface {
+	// TextMarshaler (the 'MarshalText' method) allows serialization of the client
+	// state for subsequent restoration. The marshalled value can be passed to `New`
+	// to restore Client state.
+	encoding.TextMarshaler
+
+	// Get returns a the randomness at `round` or an error.
+	// Requesting round = 0 will return randomness for the most
+	// recent known round, bounded at a minimum to the `RoundAt(time.Now())`
+	Get(ctx context.Context, round uint64) (Result, error)
+
+	// Watch returns new randomness as it becomes available.
+	Watch(ctx context.Context) <-chan Result
+
+	// RoundAt will return the most recent round of randomness that will be available
+	// at time for the current client.
+	RoundAt(time time.Time) uint64
+}
+
+// Result represents the randomness for a single drand round.
+type Result struct {
+	Round     uint64
+	Signature []byte
```

Comment:
I would like this to return the previous signature as well - otherwise we loose the property that a beacon is verifiable by itself.

Reply:
This client setup is internally doing local Verification, so the thought is that the only data emitted by it will already be validated as random. Is it still useful for the consumer above it to also be able to validate?

Reply:
Ah I see, right it's already validated. 
Well in that case the question whether the consumer needs to store/display for some reasons the previous signature ? I would tend to still include it to be on the safe side...

Reply:
see if you're happy with the current interface + concrete struct manner of addressing this. By default, you only get randomness, but it can be cast to the concrete struct if the consumer wants to access the signature. That way it requires conscious thought to use the extra fields.